src/borg/testsuite/repository.py

```diff
@@ -728,20 +728,21 @@ def test_repair_missing_segment(self):
         self.add_objects([[1, 2, 3], [4, 5, 6]])
         self.assert_equal(set([1, 2, 3, 4, 5, 6]), self.list_objects())
         self.check(status=True)
-        self.delete_segment(2)
+        self.delete_segment(3)
         self.repository.rollback()
         self.check(repair=True, status=True)
         self.assert_equal(set([1, 2, 3]), self.list_objects())
 
     def test_repair_missing_commit_segment(self):
         self.add_objects([[1, 2, 3], [4, 5, 6]])
-        self.delete_segment(3)
+        self.delete_segment(5)  # XXX problem: if the last and only commit is killed, we have to diagnose "beyond repair"
         self.assert_raises(Repository.ObjectNotFound, lambda: self.get_objects(4))
         self.assert_equal(set([1, 2, 3]), self.list_objects())
 
     def test_repair_corrupted_commit_segment(self):
         self.add_objects([[1, 2, 3], [4, 5, 6]])
-        with open(os.path.join(self.tmppath, 'repository', 'data', '0', '3'), 'r+b') as fd:
+        with open(os.path.join(self.tmppath, 'repository', 'data', '0', '5'), 'r+b') as fd:
+            # XXX problem: if the last and only commit is corrupted, we have to diagnose "beyond repair"
```

Comment:
if borg check does not find any commit tag, it currently tells "this is beyond repair".